core/modulesource.go

```diff
@@ -425,284 +279,82 @@ func (src *ModuleSource) LoadContext(ctx context.Context, dag *dagql.Server, pat
-		// context path with the module root subpath
-		moduleRootAbsPath = filepath.Join(contextAbsPath, src.AsLocalSource.Value.RootSubpath)
-	}
-
-	return contextAbsPath, moduleRootAbsPath, nil
-}
-
-// resolveContextPaths returns the context path to the .git directory
-// if it exists. Otherwise, it returns the source root directory.
-func (src *ModuleSource) ResolveContextPathFromCaller(ctx context.Context) (contextRootAbsPath, sourceRootAbsPath string, _ error) {
-	if src.Kind != ModuleSourceKindLocal {
-		return "", "", fmt.Errorf("cannot resolve non-local module source from caller")
-	}
-
-	rootSubpath, err := src.SourceRootSubpath()
-	if err != nil {
-		return "", "", fmt.Errorf("failed to get source root subpath: %w", err)
-	}
-
-	bk, err := src.Query.Buildkit(ctx)
-	if err != nil {
-		return "", "", fmt.Errorf("failed to get buildkit client: %w", err)
-	}
-	sourceRootStat, err := bk.StatCallerHostPath(ctx, rootSubpath, true)
-	if err != nil {
-		return "", "", fmt.Errorf("failed to stat source root: %w", err)
-	}
-	sourceRootAbsPath = sourceRootStat.Path
-
-	contextAbsPath, contextFound, err := callerHostFindUpContext(ctx, bk, sourceRootAbsPath)
-	if err != nil {
-		return "", "", fmt.Errorf("failed to find up root: %w", err)
-	}
-
-	if !contextFound {
-		// default to restricting to the source root dir, make it abs though for consistency
-		contextAbsPath = sourceRootAbsPath
-	}
-
-	return contextAbsPath, sourceRootAbsPath, nil
-}
-
-// context path is the parent dir containing .git
-func callerHostFindUpContext(
-	ctx context.Context,
-	bk *buildkit.Client,
-	curDirPath string,
-) (string, bool, error) {
-	stat, err := bk.StatCallerHostPath(ctx, filepath.Join(curDirPath, ".git"), true)
-	if err == nil {
-		// NOTE: important that we use stat.Path here rather than curDirPath since the stat also
-		// does some normalization of paths when the client is using case-insensitive filesystems
-		return filepath.Dir(stat.Path), true, nil
-	}
-	// TODO: remove the strings.Contains check here (which aren't cross-platform),
-	// since we now set NotFound (since v0.11.2)
-	if status.Code(err) != codes.NotFound && !strings.Contains(err.Error(), "no such file or directory") {
-		return "", false, fmt.Errorf("failed to lstat .git: %w", err)
-	}
-
-	nextDirPath := filepath.Dir(curDirPath)
-	if curDirPath == nextDirPath {
-		return "", false, nil
-	}
-	return callerHostFindUpContext(ctx, bk, nextDirPath)
-}
-
-func (src *ModuleSource) ContextDirectory() (inst dagql.Instance[*Directory], err error) {
-	switch src.Kind {
-	case ModuleSourceKindLocal:
-		if !src.AsLocalSource.Valid {
-			return inst, fmt.Errorf("local src not set")
-		}
-		if !src.AsLocalSource.Value.ContextDirectory.Valid {
-			return inst, fmt.Errorf("local src context directory not set")
-		}
-		return src.AsLocalSource.Value.ContextDirectory.Value, nil
-	case ModuleSourceKindGit:
-		if !src.AsGitSource.Valid {
-			return inst, fmt.Errorf("git src not set")
+	case ModuleSourceKindDir:
```

Comment:
Yes!! :tada: this cleans up a very nice long-standing itch for me :tada:

Reply:
Yeah pretending those were local was the epitome of hack :guyfierichefkiss:, glad it's gone now too